CMakeLists.txt

```diff
@@ -726,177 +432,145 @@ if (WIN32)
+    set(ARCH_SOURCES_DIR arm64)
+elseif (CLR_CMAKE_TARGET_ARCH_ARM)
+    set(ARCH_SOURCES_DIR arm)
+elseif (CLR_CMAKE_TARGET_ARCH_I386)
+    set(ARCH_SOURCES_DIR i386)
+else ()
+    clr_unknown_arch()
+endif ()
 
-  if (CLR_CMAKE_PLATFORM_UNIX_TARGET_AMD64)
-    add_definitions(-DUNIX_AMD64_ABI)
-  elseif (CLR_CMAKE_PLATFORM_UNIX_TARGET_ARM)
-    add_definitions(-DUNIX_ARM_ABI)
-  endif()
+# Enable for UNIX altjit on Windows - set(CLR_CMAKE_PLATFORM_UNIX_AMD64 1)
+# Enable for UNIX altjit on Windows - add_definitions(-DCLR_CMAKE_PLATFORM_UNIX=1)
 
+#--------------------------------------
+# Compile Options
+#--------------------------------------
+include(compileoptions.cmake)
+
+#----------------------------------------------------
+# Cross target Component build specific configuration
+#----------------------------------------------------
+if(CLR_CROSS_COMPONENTS_BUILD)
+  include(crosscomponents.cmake)
+endif(CLR_CROSS_COMPONENTS_BUILD)
+
+#-----------------------------------------
+# Add Projects
+#     - project which require platform header not clr's
+#     - do not depend on clr's compile definitions
+#-----------------------------------------
+if(CLR_CMAKE_PLATFORM_UNIX)
+  add_subdirectory(src/corefx)
 endif(CLR_CMAKE_PLATFORM_UNIX)
 
-add_definitions(-D_BLD_CLR)
-add_definitions(-DDEBUGGING_SUPPORTED)
-add_definitions(-DPROFILING_SUPPORTED)
+if(CLR_CMAKE_PLATFORM_UNIX)
+  add_subdirectory(src/ToolBox/SOS/lldbplugin)
+  if(NOT DEFINED CLR_CROSS_COMPONENTS_BUILD)
+    add_subdirectory(src/pal)
+  endif(NOT DEFINED CLR_CROSS_COMPONENTS_BUILD)
+  add_subdirectory(src/coreclr/hosts)
+  add_subdirectory(src/ildasm/unixcoreclrloader)
+endif(CLR_CMAKE_PLATFORM_UNIX)
 
-if(WIN32)
-  add_definitions(-DWIN32)
-  add_definitions(-D_WIN32)
-  add_definitions(-DWINVER=0x0602)
-  add_definitions(-D_WIN32_WINNT=0x0602)
-  add_definitions(-DWIN32_LEAN_AND_MEAN=1)
-  if(CLR_CMAKE_PLATFORM_ARCH_AMD64 OR CLR_CMAKE_PLATFORM_ARCH_I386)
-    add_definitions(-D_CRT_SECURE_NO_WARNINGS)
-    # Only enable edit and continue on windows x86 and x64
-    # exclude Linux, arm & arm64
-    add_definitions(-DEnC_SUPPORTED)
-  endif(CLR_CMAKE_PLATFORM_ARCH_AMD64 OR CLR_CMAKE_PLATFORM_ARCH_I386)  
-endif(WIN32)
+# Add this subdir. We install the headers for the jit.
+add_subdirectory(src/pal/prebuilt/inc)
 
-# Features - please keep them alphabetically sorted
+add_subdirectory(src/debug/debug-pal)
 
-add_definitions(-DFEATURE_APPDOMAIN_RESOURCE_MONITORING)
 if(WIN32)
-    add_definitions(-DFEATURE_APPX)
-endif(WIN32)
-if(CLR_CMAKE_PLATFORM_ARCH_AMD64 OR CLR_CMAKE_PLATFORM_ARCH_ARM OR CLR_CMAKE_PLATFORM_ARCH_ARM64)
-    add_definitions(-DFEATURE_ARRAYSTUB_AS_IL)
+  add_subdirectory(src/gc/sample)
 endif()
 
-add_definitions(-DFEATURE_ASYNC_IO)
-add_definitions(-DFEATURE_BCL_FORMATTING)
-add_definitions(-DFEATURE_COLLECTIBLE_TYPES)
-
+# Above projects do not build with these compile options
+# All of the compiler options are specified in file CompilerOptions.cmake
```

Comment:
Typo - the actual name is compileoptions.cmake (here you have compile**r**options.cmake)